LoopKit/InsulinKit/HKQuantitySample+InsulinKit.swift

```diff
@@ -57,6 +73,20 @@ extension HKQuantitySample {
         case .resume:
             return nil
         }
+        
+        // Save the insulin model
+        metadata[MetadataKeyInsulinCurveDuration] = dose.insulinModel?.effectDuration
+        metadata[MetadataKeyInsulinCurveDelay] = dose.insulinModel?.delay
+
+        if let model = dose.insulinModel as? ExponentialInsulinModel {
+            metadata[MetadataKeyInsulinCurvePeak] = model.peakActivityTime
+            metadata[MetadataKeyInsulinCurveType] = InsulinModelType.exponential.rawValue
+        } else if let model = dose.insulinModel as? ExponentialInsulinModelPreset {
+            metadata[MetadataKeyInsulinCurvePeak] = (model.getExponentialModel() as! ExponentialInsulinModel).peakActivityTime
+           metadata[MetadataKeyInsulinCurveType] = InsulinModelType.exponential.rawValue
+        } else if let _ = dose.insulinModel as? WalshInsulinModel {
+            metadata[MetadataKeyInsulinCurveType] = InsulinModelType.walsh.rawValue
```

Comment:
This is another place where I'm wondering if there's a more elegant way to assign the values.